app/modules/finance/operations_router.py

```
@@ -48,13 +52,35 @@ def create_payment_voucher(payload:dict,db:Session=Depends(get_db),principal:Pri
         if not payload.get(k):raise HTTPException(400,f'{k} is required')
     if db.query(m.FinancePaymentVoucher).filter_by(school_id=principal.school_id,voucher_number=payload['voucher_number']).first():raise HTTPException(409,'Voucher number already exists')
     r=m.FinancePaymentVoucher(school_id=principal.school_id,voucher_number=payload['voucher_number'],payee=payload['payee'],amount=_money(payload['amount']),description=payload['description'],invoice_reference=payload.get('invoice_reference'),lpo_reference=payload.get('lpo_reference'),requested_by=principal.user_id);db.add(r);_audit(db,principal,'create','payment_voucher',0,f'Created voucher {r.voucher_number}');db.commit();db.refresh(r);return r
+@router.get('/finance/payment-vouchers/{voucher_id}/approval-status')
+def payment_voucher_approval_status(voucher_id:int,db:Session=Depends(get_db),principal:Principal=Depends(require_role('viewer','admin'))):
+    r=db.query(m.FinancePaymentVoucher).filter_by(id=voucher_id,school_id=principal.school_id).first()
+    if not r: raise HTTPException(404,'Payment voucher not found')
+    required=_required_approvals(_money(r.amount)); approvals=db.query(m.FinanceApproval).filter_by(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id,decision='APPROVE').order_by(m.FinanceApproval.sequence).all()
+    return {'voucher_id':r.id,'voucher_number':r.voucher_number,'amount':_money(r.amount),'required_approvals':required,'approved_count':len(approvals),'next_sequence':len(approvals)+1 if len(approvals)<required else None,'status':r.status,'approvals':[{'sequence':a.sequence,'approver':a.approver,'decided_at':a.decided_at} for a in approvals]}
 @router.post('/finance/payment-vouchers/{voucher_id}/approve')
 def approve_payment_voucher(voucher_id:int,sequence:int=Query(1,ge=1,le=3),decision:str=Query('APPROVE'),db:Session=Depends(get_db),principal:Principal=Depends(require_role('admin'))):
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)